icu4c/source/common/rbbi_cache.cpp

```diff
@@ -420,50 +427,133 @@ UBool RuleBasedBreakIterator::BreakCache::populateNear(int32_t position, UErrorC
     return true;
 }
 
+static inline bool isNumber(UChar32 ch) {
+    static UChar32 range[2][2] = {
```

Comment:
I think you should construct a UnicodeSet (private static) in RuleBasedBreakIterator::BreakCache to test the number. The cost for doing so won't be huge.  @markusicu What is the right way to specify all number? "[:N:]" ? or should he use "[:Nd:]" ?

Reply:
Resolved. 
(Both  Nd and N could work)